pyvista/plotting/lights.py

```diff
@@ -295,15 +294,16 @@ def ambient_color(self):
         >>> light = pyvista.Light()
         >>> light.ambient_color = 'red'
         >>> light.ambient_color
-        (1.0, 0.0, 0.0)
+        Color(name='red', hex='#ff0000ff')
 
         """
-        return self.GetAmbientColor()
+        return self._ambient_color
 
     @ambient_color.setter
-    def ambient_color(self, color):
+    def ambient_color(self, color: color_like):
         """Set the ambient color of the light."""
-        self.SetAmbientColor(parse_color(color)[:3])
+        self._ambient_color = Color(color)
```

Comment:
We can just use `Color(self.GetAmbientColor())` in the getter (I did exactly that for `Renderer.background_color`), but this would indeed remove the name. I haven't really analyzed the overhead of creating a new `Color`, so I will check that first. I was thinking we could add a reverse dictionary, mapping color hex strings to color names, such that a color name is always available even if it was not created from that name (I think the overhead of an extra dict lookup would be low enough?)

Reply:
Great idea! The dict lookup is O(1) an fast so at worst the memory footprint of the reverse lookup dict would matter, but again that's not a huge dict. One potential (but not serious) issue is that if we have multiple names mapping to the same colour then the reverse mapping will obviously have to stick to one. But hopefully the names would be synonymous that this wouldn't confuse anyone. And in any case we can check if the hex codes are unique.

Regarding the overhead of creating `Color` instances: we might even consider caching instances? Odds are a given colour is used in multiple places during a single session. Of course this only works if `Color` is immutable. But I don't think I remember setters in the implementation so we might be partway there. Again this is something that can be a future PR as an optimisation.

Reply:
The duplicate names can be solved by generalizing the `color_char_to_word` dict to a general `color_synonyms` dict I think. Then we can choose an arbitrary name as the 'default', defined in `hexcolors`, and specify all synonyms in `color_synonyms`.

The `Color` class is indeed intended to be immutable, so caching could be investigated in a future PR.

Reply:
I don't think going to such measures would be worthwhile, after all in almost all cases the reverse colour mapping will work, and in rare cases of duplicate names the replacing label should in principle be synonymous with the input label. Of course if you feel like implementing it it will round out this edge case :)

Reply:
Done in last commit, wasn't too hard and provides a nice way to deal with the 'paraview' and 'pv' color synonyms.